src/Microsoft.AspNetCore.Mvc.Core/RedirectResult.cs

```diff
@@ -99,15 +105,52 @@ public string Url
         public IUrlHelper UrlHelper { get; set; }
 
         /// <inheritdoc />
+        public override Task ExecuteResultAsync(ActionContext context)
+        {
+            if (context == null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+
+            var executor = context.HttpContext.RequestServices.GetRequiredService<IActionResultExecutor<RedirectResult>>();
+            return executor.ExecuteAsync(context, this);
+        }
+
+#pragma warning disable CS0809
+        [Obsolete("This implementation will be removed in a future release, use ExecuteResultAsync.")]
         public override void ExecuteResult(ActionContext context)
         {
             if (context == null)
             {
                 throw new ArgumentNullException(nameof(context));
             }
 
-            var executor = context.HttpContext.RequestServices.GetRequiredService<RedirectResultExecutor>();
-            executor.Execute(context, this);
+            var services = context.HttpContext.RequestServices;
```

Comment:
Could we have these call the executor and block on it? In the default case, it should be fine since the executor is entirely synchronous.

Reply:
I don't want to put a `.Result` in the code. You could implement these interfaces and do some arbitrary stuff